rflx/identifier.py

```diff
@@ -1,26 +1,39 @@
-from typing import Sequence, Union
+from typing import Optional, Sequence, Union
 
 from rflx.common import generic_repr
 from rflx.contract import invariant
+from rflx.error import Location
 
 
 @invariant(lambda self: all(" " not in part for part in self.parts), "no whitespace in identifier")
 @invariant(lambda self: "" not in self.parts, "no empty part in identifier")
 class ID:
-    def __init__(self, identifier: Union[str, Sequence[str], "ID"] = None) -> None:
+    def __init__(
+        self, identifier: Union[str, Sequence[str], "ID"] = None, location: Location = None
+    ) -> None:
         self.parts: Sequence[str]
-        if isinstance(identifier, str):
+        self.location = location
+
+        if identifier is None:
+            self.parts = []
+        elif isinstance(identifier, str):
             self.parts = identifier.split(".")
         elif isinstance(identifier, list):
             self.parts = identifier
         elif isinstance(identifier, ID):
             self.parts = list(identifier.parts)
+            self.location = identifier.location
         else:
             assert False, f'unexpected identifier type "{type(identifier).__name__}"'
 
     def __eq__(self, other: object) -> bool:
         if isinstance(other, self.__class__):
-            return self.__dict__ == other.__dict__
+            return self.parts == other.parts
+        return NotImplemented
+
+    def __lt__(self, other: object) -> bool:
+        if isinstance(other, self.__class__):
+            return str(self) < str(other)
         return NotImplemented
```

Comment:
It is required in `Field.__lt__` which in turn is required to sort the paths collected in `__proof_reachability` when an unreachable field is found. Otherwise, the order of paths is not reproducible and some tests fail randomly.